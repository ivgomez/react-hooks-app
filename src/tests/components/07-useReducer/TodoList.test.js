import "@testing-library/jest-dom";
import React from "react";
import { shallow } from "enzyme";
import { demoTodos } from "../../fixtures/demoTodos";
import { TodoList } from "../../../components/07-useReducer/TodoList";

describe("Test on <TodoList />", () => {
  const handleDelete = jest.fn();
  const handleToggle = jest.fn();
  const wrapper = shallow(
    <TodoList todos={demoTodos} handleToggle={handleToggle} handleDelete={handleDelete} />
  );
  test("should rendered ok", () => {
    expect(wrapper).toMatchSnapshot();
  });
  test("should have 2 list items", () => {
    expect(wrapper.find("TodoListItem").length).toBe(demoTodos.length);
    expect(wrapper.find("TodoListItem").at(0).prop("handleDelete")).toEqual(expect.any(Function));
  });
});
