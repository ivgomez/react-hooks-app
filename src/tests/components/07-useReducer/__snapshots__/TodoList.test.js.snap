// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <TodoList /> should rendered ok 1`] = `
<ul
  className="list-group list-group-flush"
>
  <TodoListItem
    handleDelete={[MockFunction]}
    handleToggle={[MockFunction]}
    index={0}
    key="1"
    todo={
      Object {
        "desc": "Learning react",
        "done": false,
        "id": 1,
      }
    }
  />
  <TodoListItem
    handleDelete={[MockFunction]}
    handleToggle={[MockFunction]}
    index={1}
    key="2"
    todo={
      Object {
        "desc": "Learnin sass",
        "done": false,
        "id": 2,
      }
    }
  />
</ul>
`;
